docs: update releases guide for v16

jelbourn opened the pull request:
Also clarify that the deprecations guide is for _noteworthy_ deprecations and may not be comprehensive.

thesmiler commented on aio/content/guide/releases.md:
```diff
@@ -130,12 +130,13 @@ All major releases are typically supported for 18 months.
-|:---     |:---    |:---        |:---         |:---        |
-| ^15.0.0 | Active | 2022-11-18 | 2023-05-18  | 2024-05-18 |
+|:--------|:-------|:-----------|:------------|:-----------|
+| ^16.0.0 | Active | 2023-05-03 | 2023-11-03  | 2025-11-03 |
```
LTS should end 2024-11-03, one year after active neds, no?

Also, while active technically ends 6 months after release, maybe we should put 2023-11-08 (or 2023-11-10) there, otherwise people might think we have NO active release for a bit

jelbourn replied:
Yeah, this was a typo, good catch. I nudged the end date a few days ahead

thesmiler commented on aio/content/guide/releases.md:
```diff
@@ -130,12 +130,13 @@ All major releases are typically supported for 18 months.
-| ^15.0.0 | Active | 2022-11-18 | 2023-05-18  | 2024-05-18 |
+|:--------|:-------|:-----------|:------------|:-----------|
+| ^16.0.0 | Active | 2023-05-03 | 2023-11-03  | 2025-11-03 |
+| ^15.0.0 | LTS    | 2022-11-18 | 2023-05-18  | 2024-05-18 |
```
Probably active should now retroactively end 2023-05-03, as it is already LTS now, but this is a nit.

jelbourn replied:
Done

thesmiler commented on aio/content/guide/releases.md:
```diff
@@ -130,12 +130,13 @@ All major releases are typically supported for 18 months.
+| ^16.0.0 | Active | 2023-05-03 | 2023-11-03  | 2025-11-03 |
+| ^15.0.0 | LTS    | 2022-11-18 | 2023-05-18  | 2024-05-18 |
 | ^14.0.0 | LTS    | 2022-06-02 | 2022-11-18  | 2023-11-18 |
 | ^13.0.0 | LTS    | 2021-11-04 | 2022-06-02  | 2023-05-04 |
```
as of today v13 is not supported, so you can delete this line.

thesmiler commented on aio/content/guide/releases.md:
```diff
@@ -130,12 +130,13 @@ All major releases are typically supported for 18 months.
 | ^13.0.0 | LTS    | 2021-11-04 | 2022-06-02  | 2023-05-04 |
 
-Angular versions v2 to v12 are no longer under support.
+Angular versions v2 to v12 are no longer supported.
```
v13 now.

thesmiler reviewed (commented):
Some nits around support dates.

alan-agius4 commented:
We also need to update https://github.com/angular/angular/blob/main/aio/content/guide/versions.md

jelbourn committed:
docs: update releases guide for v16

Also clarify that the deprecations guide is for _noteworthy_ deprecations and may not be comprehensive.

jelbourn commented:
Comments addressed

thesmiler commented:
LGTM now

alxhub commented:
This PR was merged into the repository by commit ed7c881c1441e181fc449f6a90c4c35c594ce300.
